contentcuration/kolibri_public/import_metadata_view.py

```diff
@@ -69,6 +70,14 @@ def retrieve(self, request, pk=None):  # noqa: C901
         :return: an object with keys for each content metadata table and a schema_version key
         """
 
+        try:
+            UUID(pk)
+        except ValueError:
+            return Response(
+                {"error": "Invalid UUID format."},
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
Given that this is a retrieve operation, a `404` would be fine here.

Also, you could just put a try catch around:
```
node = get_object_or_404(models.ContentNode.objects.all(), pk=pk)
```

Catch the value error and raise a 404 there.

Reply:
Looks like in Kolibri that line was updated to use `self.get_object()` - not sure if that's catching the ValueError or not, but I suspect it may be: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/content/public_api.py#L75